projects/storefrontlib/src/cms-structure/outlet/provide-outlet.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  APP_INITIALIZER,
+  ComponentFactoryResolver,
+  FactoryProvider,
+  Type,
+} from '@angular/core';
+import { OutletPosition } from './outlet.model';
+import { OutletService } from './outlet.service';
+
+interface ProvideOutletOptions {
+  name: string;
+  position?: OutletPosition;
+  component: Type<any>;
+}
+
+export function provideOutlet(options: ProvideOutletOptions): FactoryProvider {
+  function appInitFactory(
+    componentFactoryResolver: ComponentFactoryResolver,
+    outletService: OutletService
+  ) {
+    return () => {
+      if (options.component) {
+        const factory = componentFactoryResolver.resolveComponentFactory(
+          options.component
+        );
+        outletService.add(
+          options.name,
+          <any>factory,
+          options.position ?? OutletPosition.REPLACE
```

Comment:
We could also default to `AFTER`. or do this in addition: 

```ts
export function provideOutletAfter(
  outletRef: string,
  component: Type<any>
): FactoryProvider {
  return provideOutlet({
    outletRef,
    component,
    position: OutletPosition.AFTER,
  });
}
```

Reply:
I'm not too interested in making it a public uti. After discussion with Stan I'd like to hide this `provideOutlet` util only inside the `product/configurator` lib as a private thing. 
I believe providing outlets in APP_INITIALIZER is not best thing we could do. For simplicity I'd rather do it via standard config.

Reply:
We can also treat is as an abstraction that will make transition to config driven outlets more seamless (so in future provideOutlet will just provide config, and we can either keep the util or deprecate it).

Reply:
I am interested in making this function public. It is very useful for projects, and it will help projects not to use pollute their app component with templates. 

I also like the idea of moving the config going forward in a config and let it picked up in one go.

Reply:
Note for @ChristophHi: This util has been already merged to `develop` in a separate PR.